napari/_qt/qt_main_window.py

```diff
@@ -32,14 +28,27 @@
 from .dialogs.qt_plugin_report import QtPluginErrReporter
 from .dialogs.screenshot_dialog import ScreenshotDialog
 from .perf.qt_debug_menu import DebugMenu
-from .qt_event_loop import get_app
+from .qt_event_loop import NAPARI_ICON_PATH, get_app
 from .qt_resources import get_stylesheet
 from .qt_viewer import QtViewer
 from .utils import QImg2array
 from .widgets.qt_plugin_sorter import QtPluginSorter
 from .widgets.qt_viewer_dock_widget import QtViewerDockWidget
 
 
+class _QtMainWindow(QMainWindow):
+    _window_icon = NAPARI_ICON_PATH
```

Comment:
Similarly here, I'm wondering why we're importing and storing the `NAPARI_ICON_PATH`, instead of maybe getting it from `app._icon_path` (assuming we add something like that in `get_app`, and then pass that to this function. While it's not necessary I think I would find that a bit simpler from a code perspective.

Reply:
I did that as a convenience for @Czaki... If he wants to, in his application, he can patch `napari._qt.qt_main_window._QtMainWindow._window_icon` to whatever string he wants to, and then that will be the icon used (rather than him needing to call `viewer.window._qt_window.setWindowIcon` after every instantiation).  (this gives it a safe namespace that can be modified without changing our internal `NAPARI_ICON_PATH`)

Reply:
Ok I get it. That's fine. Maybe you can add a comment here like
```
# This was added so that someone can patch `napari._qt.qt_main_window._QtMainWindow._window_icon` to
# whatever string they want, and that icon will always be used, and so
# `viewer.window._qt_window.setWindowIcon` won't need to be called after every instantiation.

Reply:
added